source.go

```diff
@@ -40,16 +41,34 @@ func (p position) toSDKPosition() (sdk.Position, error) {
 	return bytes, nil
 }
 
+func fromSDKPosition(sdkPos sdk.Position) (position, error) {
+	pos := position{}
+	err := json.Unmarshal(sdkPos, &pos)
+	if err != nil {
+		return position{}, fmt.Errorf("failed unmarshalling position: %w", err)
+	}
+	return pos, nil
+}
+
 type recordPayload struct {
 	Plaintext string            `json:"plaintext"`
 	Metadata  map[string]string `json:"metadata"`
```

Comment:
Sidenote: I assume this was done because it's an v0.2.0 connector and the record metadata otherwise wouldn't be produced. When we move to the new SDK I recommend removing this and using record metadata instead.

Reply:
Correct and because @samirketema  in his project needs the metadata in the destination, and this was quicker for him to use (with metadata being separate he'd need a transform to merge it into the payload, so we did the work here to unblock him).